pkg/cluster/k8sres.go

```diff
@@ -1000,20 +1010,31 @@ func (c *Cluster) getPodEnvironmentSecretVariables() ([]v1.EnvVar, error) {
 	}
 
 	for k := range secret.Data {
-		secretPodEnvVarsList = append(secretPodEnvVarsList,
-			v1.EnvVar{Name: k, ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: &v1.SecretKeySelector{
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: c.OpConfig.PodEnvironmentSecret,
+		if !isEnvVarPresent(secretPodEnvVarsList, k) {
+			secretPodEnvVarsList = append(secretPodEnvVarsList,
+				v1.EnvVar{Name: k, ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: c.OpConfig.PodEnvironmentSecret,
+						},
+						Key: k,
 					},
-					Key: k,
-				},
-			}})
+				}})
+		}
 	}
 
 	return secretPodEnvVarsList, nil
 }
 
+func isEnvVarPresent(envs []v1.EnvVar, key string) bool {
+	for _, env := range envs {
+		if env.Name == key {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Actually, it might be even better to do `isEnvVarPresent` in [generateSpiloPodEnvVars](https://github.com/zalando/postgres-operator/blob/43e18052c427e255afaa985a046130fc47cf9264/pkg/cluster/k8sres.go#L880), too, instead of appending duplicate variables and resolving it later with `deduplicateEnvVars`.

Reply:
Fix logic, please review.